crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -746,8 +746,10 @@ def nodes(component = "all")
         end
 
         # if all nodes are upgraded, cleanup and end the whole step
-        status = ::Crowbar::UpgradeStatus.new
-        if status.progress[:remaining_nodes].zero?
+#        status = ::Crowbar::UpgradeStatus.new
+#        if status.progress[:remaining_nodes].zero?
+# FIXME with some nodes being shut down and others upgraded, remaining_nodes should return 0...
```

Comment:
Could you clarify on this? What is the difference that is relevant to this change? AFAICS status.progress[:remaining_nodes] is initialized from remaining_nodes() at the beginning of the update (At which point it will be the number of node in the "crowbar_upgrade" state, which doesn't include nodes that are powered off). Then it will be decreased every time a node transitions to "upgraded". Is that exactly as "remaining_nodes" behaves or am I missing something?

Reply:
Hm, looks like you are right. When `status.progress[:remaining_nodes]` is initialized, nodes that are shut down are already ignored.